trail/upload/models.py

```diff
@@ -481,14 +482,127 @@ def toDict(self):
+        and then return them as a list of `Thumbnails` object(s).
+
+        Parameters
+        ----------
+        data : `dict`
+            Dictionary containing at least the required standardized keys for
+            `Metadata`, `Wcs` and `Thumbnail`.
+
+        Returns
+        -------
+        thumbs : `list`
+            A list of one or more `Thumbnails` objects.
+        """
+        if isinstance(data, Thumbnails):
+            return [data, ]
+
+        # assume all of them are Thumbnails in a list and return
+        try:
+            if isinstance(data[0], Thumbnails):
+                return data
+        except TypeError:
+            pass
+
+        i = 0
+        thumbnails = None
+        while (thumbnails is None and i < len(cls._thumbkeys)):
+            thumbnails = data.pop(cls._thumbkeys[i])
+            i += 1
+
+        # try one more time with a flat dict, just in case someone was too lazy
+        # Otherwise, thumbnails were retrieved as a dict, or a list of dicts
+        if thumbnails is None:
+            large = data.pop("large", None)
+            small = data.pop("small", None)
+            thumbs = [Thumbnails(large=large, small=small), ]
+        elif isinstance(thumbnails, dict):
+            thumbs = [Thumbnails(large=large, small=small), ]
+        else:
+            thumbs = []
+            for thumb in thumbnails:
+                thumbs.append(Thumbnails(large=thumb["large"], small=thumb["small"]))
+
+        return thumbs
+
+    @classmethod
+    def fromDict(cls, data):
+        """Construct an StandardizedResult from a dictionary.
+
+        The dictionary can contain either a flattened set of values of a single
+        result f.e.:
+
+            {obs_lon: ... <metadata keys>, wcs_radius: ... <wcs_keys>, large: ... <thumbnail_keys>}
+
+        or have separated metadata, wcs and thumbnail keys like:
+
+            {metadata: {...}, wcs: {...}, thumbnails: {...}}
+
+        in which case the wcs and thumbnails can be iterable, i.e.
+
+            {metadata: {...}, wcs: [{...}, {...}, ... ],  thumbnails: [{...}, {...}, ... ],}
+
+        Parameters
+        ----------
+        data : `dict`
+            Dictionary containing at least the required standardized keys for
+            `Metadata`, `Wcs` and `Thumbnail`.
+        """
+        thumbs = cls.__resolve_thumbs(data)
+        header = StandardizedHeader.fromDict(data)
+        return cls(header=header, thumbnails=thumbs)
+
+    def __eq__(self, other):
+        return self.header.isClose(other)
 
-        # TODO: this doesn't work, why?
-        # Wcs.objects.bulk_create(self.wcs)
+    @property
+    def wcs(self):
+        """Standardized wcs."""
+        return self.header.wcs
+
+    @property
+    def metadata(self):
+        """Standardized metadata."""
+        return self.header.metadata
+
+    @property
+    def isMultiExt(self):
+        """True when the header is a multi extension header."""
+        return len(self.wcs) > 1
```

Comment:
I've been thinking about that myself. Wcs are really the "property" of an detector and kind of describes the mapping between what that detector sees and the real life geometry. The only thing we are interested in from a detector is the image. So each image would have a WCS. Or at least that was my logic. 

That means that it's not possible to have an image of the sky and not have a Wcs, or if we do - we don't want that image because we don't know what we're looking at. It is possible, however, to have multiple images but only 1 Wcs -> i.e. mask planes etc...

This here however, was copied over from `fits_processor`, where the above is all applied and determined, but this here has nothing to do with that. It just checks if there are multiple Wcs's in the `self.wcs` list. It's just for simplicity really, I thought it neater to ask this than to ask for `if len(stdres.wcs)==1` when using it. Will fit docstring.